drag_and_drop_v2/drag_and_drop_v2.py

```diff
@@ -666,9 +682,10 @@ def _mark_complete_and_publish_grade(self):
 
         # There's no going back from "completed" status to "incomplete"
         self.completed = self.completed or self._is_answer_correct() or not self.attempts_remain
-        grade = self._get_grade()
+        grade = self._calculate_grade()
         # ... and from higher grade to lower
-        if grade > self.grade:
+        current_grade = self._get_grade_if_set()
+        if current_grade is None or grade > current_grade:
```

Comment:
Just to follow up on this - from a product perspective, I'm OK with keeping the max score as it was designed. I think this benefits learners. We haven't heard feedback from partners/learners thus far, but we can certainly reconsider in the future. Agreed that if this changes in the future, it would be a separate story.